app/controllers/webhook_controller.rb

```diff
@@ -1,20 +1,22 @@
 class WebhookController < ApplicationController
   skip_before_action :verify_authenticity_token
+  before_action :format_params
 
   def handle
-    headers = request.headers
-    event = headers['X-GitHub-Event']
-    signature = headers['X-Hub-Signature']
-    # this is needed because the action attr is replaced by rails' action param
-    # and is lost on the params hash
-    payload = request.raw_post
-    case GithubHandler.new(event, payload, signature).handle
-    when GithubHandler::SUCCESS
-      head :ok
-    when GithubHandler::FORBIDDEN
-      head :forbidden
-    when GithubHandler::BAD_REQUEST
-      head :bad_request
-    end
+    Event.resolve(@payload) if webhook_verified?
+  end
+
+  private
+
+  def format_params
```

Comment:
I think we can come up with a better name for this. What we're really doing here is setting up the payload. Maybe we can continue with the handling of the event only if it's verified.

What do you think about something like...


```ruby
def set_payload
  event = request.headers['X-GitHub-Event']
  signature = request.headers['X-Hub-Signature']

  payload = JSON.parse(request.raw_post).merge(event: event)

  head 400 and return unless payload_verified?(payload, signature)
end

```

One more thing. Shouldn't `handle` return, for example, `head :ok`?

Reply:
Yes! I had it in that way i don't know why did i change it. The head :ok and head 400 looks good, adding...